### ISSUE TITLE
finetuning : checkpoint 0.5.1 error model shape does not match

### ISSUE THREAD

**Issue by vigneshgig** [mozilla/DeepSpeech#2278]
Instructions for updating:
Use standard file APIs to check for files with this prefix.
I0730 19:08:56.306705 139864493074176 saver.py:1280] Restoring parameters from ./deepspeech-0.5.1-checkpoint/model.v0.5.1
E Restoring from checkpoint failed. This is most likely due to a mismatch between the current graph and the graph from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:
E 
E No OpKernel was registered to support Op 'CudnnRNNCanonicalToParams' used by node tower_0/cudnn_lstm/cudnn_lstm/CudnnRNNCanonicalToParams (defined at DeepSpeech.py:115) with these attrs: [input_mode="linear_input", T=DT_FLOAT, direction="unidirectional", rnn_mode="lstm", seed2=237, dropout=0, seed=4568, num_params=8]
E Registered devices: [CPU, XLA_CPU, XLA_GPU]
E Registered kernels:
E   device='GPU'; T in [DT_DOUBLE]
E   device='GPU'; T in [DT_FLOAT]
E   device='GPU'; T in [DT_HALF]
E 
E 	 [[tower_0/cudnn_lstm/cudnn_lstm/CudnnRNNCanonicalToParams]]
E The checkpoint in ./deepspeech-0.5.1-checkpoint/model.v0.5.1 does not match the shapes of the model. Did you change alphabet.txt or the --n_hidden parameter between train runs using the same checkpoint dir? Try moving or removing the contents of ./deepspeech-0.5.1-checkpoint/model.v0.5.1.

 python3.7 DeepSpeech.py --n_hidden 2048 --checkpoint_dir ./deepspeech-0.5.1-checkpoint --epochs 100 --train_files ./data/deepspeech_dataset/train_data_english.csv --dev_files ./data/deepspeech_dataset/validation_data_english.csv --test_files ./data/deepspeech_dataset/validation_data_english.csv --learning_rate 0.0001 --train_batch_size 200 --test_batch_size 100 --dev_batch_size 100 --export_dir ./export --use_cudnn_rnn true

I didnt include environment version because I tried without checkpoint its working but when i add the checkpoint for finetuning the my own dataset it's showing a above error

Thanks

**Comment by vigneshgig** [comment]
I download the checkpoint 0.5.1 source code Now it working
